src/GpuLidarSensor.cc

```diff
@@ -129,6 +132,15 @@ bool GpuLidarSensor::Load(const sdf::Sensor &_sdf)
     return false;
   }
 
+  if (_sdf.Element()->HasElement("scanning_pattern"))
+  {
+    auto pattern = _sdf.Element()->Get<std::string>("scanning_pattern");
+    if (pattern == "rasterizing")
```

Comment:
This should look up the correct enum value from the std::map above.
If the string is not in the map then we should log an error and return false.